src/python/WMCore/MicroService/MSUnmerged/MSUnmerged.py

```diff
@@ -431,15 +431,32 @@ def consRecordAge(self, rse):
         isConsNewer = self.rseConsStats[rseName]['end_time'] > self.rseTimestamps[rseName]['prevStartTime']
         if not isConsNewer:
             msg = "RSE: %s With old consistency record in Rucio Consistency Monitor. " % rseName
-            msg += "Skipping it in the current run."
-            self.logger.info(msg)
-            raise MSUnmergedPlineExit(msg)
+            if rse['isClean']:
```

Comment:
With the current definition of `isClean`, doesn't it mean that the service will keep trying to delete files until it manages to delete every single file that is listed as deletable? I'm afraid we need to redefine how we consider an RSE as clean or not.

Another option, which is likely better, would be to add a new key/value pair to the MSUnmergedRSE object to hold the original timestamp - when the file dump was collected - provided by Rucio ConMon. Such that we end up with an expression like:
```
isConsNewer = self.rseConsStats[rseName]['end_time'] > self.rseTimestamps[rseName]['dumpTStamp']
```

Reply:
With the current definition of `isClean` we refresh/consider as new the RSE cleanup status on every RucioConMon polling cycle. Meaning if RucioConMon provides a new dump of the RSE, we reset everything - counters and cleanup status (keeping only the timestamps/updating them as usual), and we retry a fresh cleanup regarding the new RSE situation. So if there have been some leftovers from previous Runs those must have been detected by RucioConMon and will get into the our next iteration. If we have cleaned the RSE last time. We should consider it unclean now and retry again - in case there is nothing to be deleted it will enter immediately in clean status, if there is something new - it will be deleted together with leftovers (if any).

The idea of preserving the RucioConMon timestamp is not bad, let me check what I can do to put it there.

Reply:
Hi @amaltaro, I did implement this and as simple as it sounds from the line you suggested, at the end it became a quite horrible change, because I also had to check the status of the RucioConMon record for the RSE not only here  but also in the `updateRSETimestamps` method, so we can be sure to update the record for this timestamp in MongoDB only if the RucioConMon has done a complete and successful scan of the RSE, basically re implementing the whole `consRecordAge` method, and at the end we will get exactly the same result. I think this is going to make things more error prone than simplifying them.